### ISSUE TITLE
PHAR prevents loading of some project's bootstrap files

### ISSUE THREAD

**Issue by xJakub** [vimeo/psalm#4628]
Composer assigns a hash to some autoload files, like `symfony/polyfill-php80/bootstrap.php`, to prevent them from being loaded twice. When box creates the PHAR, it adds a namespace to the code, but the hash in composer stays the same and only one of the two files will be autoloaded (under normal circumstances, PHAR's one):

This causes that, when using functions from some libraries (like symfony/polyfill-php80), you get an UndefinedFunction error:
```
ERROR: UndefinedFunction - src/test:x:13 - Function str_starts_with does not exist (see https://psalm.dev/021)
```

And the following entry is in both the project's and PHAR's `vendor/composer/autoload_files.php`:
```
'a4a119a56e50fbb293281d9a48007e0e' => $vendorDir . '/symfony/polyfill-php80/bootstrap.php',
```

I hit this bug using psalm/phar 3.18.2, and with composer 1.10.7 and 2.0.7. A quick solution would be to empty `$GLOBALS['__composer_autoload_files']` before loading the project's files, but it might be better to make box change those hashes when building the phar.

**Comment by ondrejmirtes** [comment]
Yeah, PHPStan does this to prevent this situation: https://github.com/phpstan/phpstan-src/blob/ef84d9444f889ac5606c37ce80d2fa5f19072cb8/bin/phpstan#L22-L37

**Comment by theofidry** [comment]
Related to https://github.com/humbug/php-scoper/issues/298; This is more a scoping issue than a PHAR one AFAICS
